impl/sql/mutationstorage/queue.go

```diff
@@ -34,23 +37,67 @@ func (m *Mutations) Send(ctx context.Context, domainID string, update *pb.EntryU
 	if err != nil {
 		return err
 	}
-	writeStmt, err := m.db.Prepare(insertQueueExpr)
+	return m.send(ctx, domainID, mData, time.Now())
+}
+
+// ts must be greater than all other timestamps currently recorded for domainID.
+func (m *Mutations) send(ctx context.Context, domainID string, mData []byte, ts time.Time) error {
+	tx, err := m.db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
 	if err != nil {
 		return err
 	}
-	defer writeStmt.Close()
-	_, err = writeStmt.ExecContext(ctx, domainID, time.Now().UnixNano(), mData)
-	return err
+
+	var maxTime int64
+	if err := tx.QueryRowContext(ctx,
+		`SELECT COALESCE(MAX(Time), 0) FROM Queue WHERE DomainID = ?;`,
```

Comment:
I'm gradually moving code away from the constants. I found the in-line statements a bit easier to read - what do you think? 

I'm also moving away from prepared statements.  I was security-trained to always prepare statements to avoid SQL injection, but the Go APIs are actually quite safe by default - ie, we're not dynamically building a query string)

Reply:
I also find in-line statements easier to follow (and I think it's easier to see that argument counts match up), so SGTM.